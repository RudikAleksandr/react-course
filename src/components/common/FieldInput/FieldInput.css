input.form-control,
input.form-control:focus {
  display: initial;
  height: 50px;
  color: white;
  font-size: 1.2rem;
  background-color: #424242;
  border-color: #424242;
}

input.form-control::-webkit-outer-spin-button,
input.form-control::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
}
