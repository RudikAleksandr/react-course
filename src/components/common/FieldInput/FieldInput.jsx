import React from 'react';
import { FormControl, FormGroup } from 'react-bootstrap';
import Label from '../Label';
import './FieldInput.css';

const FieldInput = ({ label, ...props }) => (
  <FormGroup>
    {label && <Label text={label} />}
    <FormControl type="calendar" {...props} />
  </FormGroup>
);

export default FieldInput;
