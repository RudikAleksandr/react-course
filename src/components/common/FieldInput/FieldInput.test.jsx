import React from 'react';
import { render } from '@testing-library/react';
import FieldInput from './FieldInput';

describe('FieldInput component', () => {
  test('render FieldInput snapshot', () => {
    const component = <FieldInput label="label" />;
    const { asFragment } = render(component);

    expect(asFragment(component)).toMatchSnapshot();
  });
});
