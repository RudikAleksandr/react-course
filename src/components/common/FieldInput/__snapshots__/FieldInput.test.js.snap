// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldInput component render FieldInput snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <label
      class="label  form-label"
    >
      label
    </label>
    <input
      class="form-control"
    />
  </div>
</DocumentFragment>
`;
